### PR TITLE
chore(compass-components, home): move all compass components provider into a single component

### PR DESCRIPTION
**Pull request by gribnoysup**
I'm splitting this patch out of compass-web PR as it's fairly independant, should make reviewing both easier.

This PR moves all compass-components providers into a single one that is then used in home. We will need to do literally the same setup in compass-web, so it's convenient to have them all combined in one like that

It's a bit annoying to review because stuff in home moved around, but not much was actually changed: there is only one provider import instead of multiple ones and all the theme logic was also removed as it's all encapsulated by this new provider now.

### CODE REVIEW

**Review by Anemy — APPROVED** [review]
Nice, I've was thinking it would be nice to structure that home.tsx file nicer for a bit now, cool to see it done! Changes look good, I haven't tested locally though.

**Comment by gribnoysup** [comment]
@Anemy completely forgot this quirk of how leafygreen handles the popover props, so had to address this just now (and force pushed, sorry), can you take another look?

File: packages/compass-components/src/components/compass-components-provider.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        scrollContainerRef: React.Ref<HTMLElement>;
+      }) => React.ReactElement | null);
+} & {
+  onNextGuideGue?: GuideCueProviderProps['onNext'];
+  onNextGuideCueGroup?: GuideCueProviderProps['onNextGroup'];
+} & React.ComponentProps<typeof SignalHooksProvider>;
+
+const darkModeMediaQuery = (() => {
+  return typeof window !== 'undefined' && window.matchMedia
+    ? window.matchMedia('(prefers-color-scheme: dark)')
```
Comment (gribnoysup) [inline packages/compass-components/src/components/compass-components-provider.tsx:44]: This works well for the electron and web and allows to remove dependency on electron apis, even though we wouldn't really need it for web where I expect mms to be passing a `darkMode` prop that will switch this component to "controlled" mode

File: packages/compass-components/src/components/compass-components-provider.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+                    darkMode,
+                    portalContainerRef: setPortalContainer,
+                    scrollContainerRef: setScrollContainer,
+                  })
+                : children}
+            </ToastArea>
+          </ConfirmationModalArea>
+        </SignalHooksProvider>
+      </GuideCueProvider>
+    </LeafyGreenProvider>
```
Comment (addaleax) [inline packages/compass-components/src/components/compass-components-provider.tsx:130]: Another very optional suggestion, maybe there could be a comment describing a rule of thumb regarding which providers go into this component and which don’t? For example, my gut feeling is that we’d always want to keep the service providers for plugins separate, but that might be wrong and different people might have different gut feelings :)

File: configs/mocha-config-compass/register/jsdom-extra-mocks-register.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+// Not implemented in jsdom
+if (!window.matchMedia) {
+  window.matchMedia = global.matchMedia = (media) => {
```
[the remark this answers is not in the document]
Comment (gribnoysup) [reply configs/mocha-config-compass/register/jsdom-extra-mocks-register.js:3]: Makes sense, will update!

File: packages/compass-components/src/components/compass-components-provider.tsx (hunk 2 above)
Comment (gribnoysup) [reply packages/compass-components/src/components/compass-components-provider.tsx:130]: Totally, let me extend the current component description

Comment (gribnoysup) [reply packages/compass-components/src/components/compass-components-provider.tsx:139]: Updated the description in https://github.com/mongodb-js/compass/pull/5257/commits/01381da15275b76366aed2e177f6570adee69bb3, how does it look?

**Review by addaleax — APPROVED** [review]

**Pull request merged**